### PR TITLE
VB ConvertTupleToStruct should not include NullableAttributes in it

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/64639
Fixes https://github.com/dotnet/roslyn/issues/46291

Working on test now.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@genlu  can you ptal?

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/VisualBasicCodeRefactoringVerifier`1+Test.cs [hunk 1]
```diff
@@ -70,12 +73,30 @@ public Test()
                 set => _sharedState.EditorConfig = value;
             }
 
+            /// <summary>
+            /// The set of code action <see cref="CodeAction.Title"/>s offered the user in this exact order.
+            /// Set this to ensure that a very specific set of actions is offered.
+            /// </summary>
+            public string[]? ExactActionSetOffered { get; set; }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/VisualBasicCodeRefactoringVerifier`1+Test.cs:80]: added these to be in sync with C# refactoring verifier (and the test updates need these).

File: src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb [hunk 2]
```diff
@@ -5,28 +5,50 @@
...
 Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.ConvertTupleToStruct
+    <UseExportProvider>
     <Trait(Traits.Feature, Traits.Features.CodeActionsConvertTupleToStruct)>
     Public Class ConvertTupleToStructTests
-        Inherits AbstractVisualBasicCodeActionTest
 
-        Protected Overrides Function CreateCodeRefactoringProvider(workspace As Workspace, parameters As TestParameters) As CodeRefactoringProvider
-            Return New VisualBasicConvertTupleToStructCodeRefactoringProvider()
-        End Function
+        Private Shared Async Function TestAsync(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb:23]: moved this entire test file to the new sdk testing system.

File: src/Workspaces/CSharp/Portable/CodeGeneration/AttributeGenerator.cs [hunk 3]
```diff
@@ -81,30 +81,6 @@ internal static class AttributeGenerator
                 : null;
         }
 
-        private static bool IsCompilerInternalAttribute(AttributeData attribute)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeGeneration/AttributeGenerator.cs:84]: moved to common location so VB can use.

File: src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb [hunk 4]
```diff
@@ -5,28 +5,50 @@
...
+            Optional testHost As TestHost = TestHost.InProcess,
+            Optional actions As String() = Nothing) As Task
 
-        Protected Overrides Function MassageActions(actions As ImmutableArray(Of CodeAction)) As ImmutableArray(Of CodeAction)
-            Return FlattenActions(actions)
+            If index <> 0 Then
+                Assert.NotNull(equivalenceKey)
+            End If
+
+            ' Options ??= new OptionsCollection(LanguageNames.CSharp);
```
Comment (genlu) [inline src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb:35]: Delete

File: src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb [hunk 5]
```diff
@@ -1386,86 +1421,31 @@ Friend Structure NewStruct1
     End Operator
 End Structure
 "
-            Await TestInRegularAndScriptAsync(text, expected, testHost:=host)
-        End Function
-
-        <Theory(Skip:="https://github.com/dotnet/roslyn/issues/46291"), CombinatorialData>
-        Public Async Function TestDuplicatedName(host As TestHost) As Task
```
Comment (genlu) [inline src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb:1393]: Huh, is this sceanrio still supported? (or was it before?)

**Review by genlu — APPROVED** [review]

File: src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/VisualBasicTest/ConvertTupleToStruct/ConvertTupleToStructTests.vb]: done.

**Pull request merged**